tests/integration/test_multi_input.py

```diff
@@ -0,0 +1,195 @@
+    pointcloud: Annotated[list[Tensor], Field(min_length=1)]
+
+
+class DisparityCustomMultiInputProtocol(BaseProtocol):
+    disparity: Annotated[list[Tensor], Field(min_length=1)]
+
+
+class MultiInputTestBaseNode(BaseNode):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.scalar = Parameter(torch.tensor(1.0), requires_grad=True)
+
+    def forward(self, inputs):
+        return [self.scalar * inp for inp in inputs]
+
+    def unwrap(self, inputs: list[dict[str, list[Tensor]]]):
+        return [item for inp in inputs for key in inp for item in inp[key]]
+
+
+class FullBackbone(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [FullCustomMultiInputProtocol]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+
+class RGBDBackbone(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [RGBDCustomMultiInputProtocol]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+
+class PointcloudBackbone(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [PointcloudCustomMultiInputProtocol]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+
+class FusionNeck(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [
+            DisparityCustomMultiInputProtocol,
+            FeaturesProtocol,
+            FeaturesProtocol,
+        ]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+
+class FusionNeck2(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [FeaturesProtocol, FeaturesProtocol, FeaturesProtocol]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+
+class CustomSegHead1(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [FeaturesProtocol]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+    def wrap(self, outputs: list[Tensor]):
+        return {"segmentation": outputs}
+
+
+class CustomSegHead2(MultiInputTestBaseNode):
+    def __init__(self, **kwargs):
+        in_protocols = [
+            DisparityCustomMultiInputProtocol,
+            FeaturesProtocol,
+            FeaturesProtocol,
+        ]
+        super().__init__(**kwargs)
+        self.in_protocols = in_protocols
+
+    def wrap(self, outputs: list[Tensor]):
+        return {"segmentation": outputs}
+
+
+@pytest.fixture(scope="function", autouse=True)
+def clear_output():
+    shutil.rmtree("output", ignore_errors=True)
+
+
+@pytest.mark.parametrize(
+    "config_file", [path for path in os.listdir("configs") if "multi_input" in path]
+)
+def test_sanity(config_file):
+    # opts = [
+    #     "trainer.epochs",
+    #     "3",
+    #     "trainer.validation_interval",
+    #     "3",
+    # ]
+
+    Trainer(f"configs/{config_file}").train()
+
+    # TODO add export and eval tests
```

Comment:
Does the export and eval not work yet?

Reply:
Fixed, I forgot to add the remaining tests.